kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/jms/JmsResponseHandlerIntegrationTest.java

```diff
@@ -138,15 +143,33 @@ public void testGetTasksUseOfAsyncResponseHandlerWithMarshaller() throws Excepti
         testGetTaskResponseHandler(new AsyncResponseHandler(callback));
         // now let's check if response has arrived
         Long processInstanceId = callback.get(Long.class);
-        assertNotNull(processInstanceId);
-        assertTrue(processInstanceId.longValue() > 0);
+        assertThat(processInstanceId).isNotNull().isPositive();
 
         TaskSummaryList taskSummaryList = callback.get(TaskSummaryList.class);
-        assertNotNull(taskSummaryList);
+        assertThat(taskSummaryList).isNotNull();
+        assertThat(taskSummaryList.getItems()).isNotNull().hasSize(1);
+    }
+
+    @Test
+    public void testStartAndCompleteTaskUseOfFireAndForgetResponseHandler() {
+        testStartAndCompleteTask(new FireAndForgetResponseHandler());
+    }
+
+    @Test
+    public void testStartAndCompleteTaskUseOfAsyncResponseHandler() {
+        ResponseCallback callback = new BlockingResponseCallback(null);
+        testStartAndCompleteTask(new AsyncResponseHandler(callback));
+    }
+
+    @Test
+    public void testQueryRegistrationUseOfFireAndForgetReponseHandler() {
+        testQueryRegistration(new FireAndForgetResponseHandler());
+    }
 
-        List<TaskSummary> tasks = taskSummaryList.getItems();
-        assertNotNull(tasks);
-        assertEquals(1, tasks.size());
+    @Test
+    public void testQueryRegistrationUseOfAsyncReponseHandler() {
+        ResponseCallback callback = new BlockingResponseCallback(null);
```

Comment:
I think there is no point in doing this since all the operations executed in this test that use `AsyncResponseHandler` return `null`.